proxy/proxy.go

```diff
@@ -408,10 +408,11 @@ func (dc *skipperDialer) DialContext(ctx stdlibcontext.Context, network, addr st
 			dialingFailed: true, // indicate error happened before http
 		}
 	} else if cerr := ctx.Err(); cerr != nil {
-		// deadline exceeded or canceled in stdlib
+		// deadline exceeded or canceled in stdlib, client closed request
+		// see https://github.com/zalando/skipper/issues/687#issuecomment-405557503
 		return nil, &proxyError{
 			err:  cerr,
-			code: http.StatusGatewayTimeout,
+			code: 499,
```

Comment:
this is not a standard code. Do we have other options?

Reply:
This is used by nginx (and I think apache, but not sure).

Reply:
maybe 0? indicating that the request was not served, therefore there is no status code at all?

Reply:
0 is specially handled in skipper and returns in the end 500 @aryszka